alicloud/data_source_alicloud_db_instances.go

```diff
@@ -0,0 +1,282 @@
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Computed: true,
+						},
+						"vpc_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"vswitch_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudDBInstancesRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).rdsconn
+
+	args := rds.CreateDescribeDBInstancesRequest()
+
+	args.RegionId = string(getRegion(d, meta))
+	args.Engine = d.Get("engine").(string)
+	args.DBInstanceStatus = d.Get("status").(string)
+	args.DBInstanceType = d.Get("instance_type").(string)
+	args.VpcId = d.Get("vpc_id").(string)
+	args.VSwitchId = d.Get("vswitch_id").(string)
+	args.ConnectionMode = d.Get("connection_mode").(string)
+	args.Tags = d.Get("tags").(string)
+	args.PageSize = requests.NewInteger(PageSizeLarge)
+
+	var dbi []rds.DBInstance
+
+	var nameRegex *regexp.Regexp
+	if v, ok := d.GetOk("name_regex"); ok {
+		if r, err := regexp.Compile(v.(string)); err == nil {
+			nameRegex = r
+		}
+	}
+
+	for {
+		resp, err := conn.DescribeDBInstances(args)
+		if err != nil {
+			return err
+		}
+
+		if resp == nil || len(resp.Items.DBInstance) < 1 {
+			break
+		}
+
+		for _, item := range resp.Items.DBInstance {
+			if nameRegex != nil {
+				if !nameRegex.MatchString(item.DBInstanceDescription) {
+					continue
+				}
+			}
+			dbi = append(dbi, item)
+		}
+
+		if len(resp.Items.DBInstance) < PageSizeLarge {
+			break
+		}
+
+		args.PageNumber = args.PageNumber + requests.NewInteger(1)
+	}
+
+	return rdsInstancesDescription(d, dbi)
+}
+
+func rdsInstancesDescription(d *schema.ResourceData, dbi []rds.DBInstance) error {
+	var ids []string
+	var s []map[string]interface{}
+
+	for _, item := range dbi {
+		mapping := map[string]interface{}{
+			"id":                    item.DBInstanceId,
+			"name":                  item.DBInstanceDescription,
+			"charge_type":           item.PayType,
+			"instance_type":         item.DBInstanceType,
+			"region_id":             item.RegionId,
+			"create_time":           item.CreateTime,
+			"expire_time":           item.ExpireTime,
+			"status":                item.DBInstanceStatus,
+			"engine":                item.Engine,
+			"engine_version":        item.EngineVersion,
+			"net_type":              item.DBInstanceNetType,
+			"connection_mode":       item.ConnectionMode,
+			"instance_class":        item.DBInstanceClass,
+			"availability_zone":     item.ZoneId,
+			"master_instance_id":    item.MasterInstanceId,
+			"guard_instance_id":     item.GuardDBInstanceId,
+			"temp_instance_id":      item.TempDBInstanceId,
+			"readonly_instance_ids": item.ReadOnlyDBInstanceIds.ReadOnlyDBInstanceId,
+			"vpc_id":                item.VpcId,
+			"vswitch_id":            item.VSwitchId,
+		}
+
+		log.Printf("alicloud_db_instances - adding rds instance: %v", mapping)
```

Comment:
line 267 will output too many logs and there is no need to do it. Please remove it.

Reply:
done